spec/requests/api/member_services/members_spec.rb

```diff
@@ -4,110 +4,96 @@
-          put '/api/member_services/members/', params: params, headers: valid_headers
-          expect(response.status).to eq 200
-          expect(member.reload.updated_at).to be_within(0.1.seconds).of(Time.now)
-          expect(json_hash.with_indifferent_access).to match(member: {
-            id: 3,
-            email: 'test@sumofus.org',
-            first_name: 'Max',
-            last_name: 'Testy-Smith',
-            country: 'United Kingdom',
-            postal: 'EC2 1AB',
-            city: 'Brussels',
-            title: 'miss',
-            address1: 'Some street',
-            address2: 'Some address',
-            created_at: /\A\d{4}(-\d{2}){2}T(\d{2}:){2}\d{2}[.]\d{3}[a-zA-Z]\z/,
-            updated_at: /\A\d{4}(-\d{2}){2}T(\d{2}:){2}\d{2}[.]\d{3}[a-zA-Z]\z/,
-            actionkit_user_id: 'mega_akid',
-            donor_status: 'donor',
-            more: '{}'
-          })
-        end
+        }
+      end
 
-        context 'when a member with the given email address does not exist' do
-          let(:params) do
-            {
-              email: 'new_member@sumofus.org',
-              first_name: 'Max',
-              last_name: 'Testy-Smith',
-              country: 'United Kingdom',
-              postal: 'EC2 1AB'
-            }
-          end
+      it 'updates the member details' do
+        put '/api/member_services/members/', params: params, headers: headers
+        expect(response).to have_http_status(:ok)
+        member.reload
+        expect(member.email).to      eq('changed@sumofus.org')
+        expect(member.first_name).to eq('Max')
+        expect(member.last_name).to  eq('Testy-Smith')
+        expect(member.country).to    eq('United Kingdom')
+        expect(member.postal).to     eq('EC2 1AB')
+      end
 
-          it 'logs error and sends back 404' do
-            error_body = {
-              errors:  [
-                'No member associated with email address new_member@sumofus.org.'
-              ]
-            }
+      it "returns the member's json object" do
+        put '/api/member_services/members/', params: params, headers: headers
+        expect(json_hash).to match('member' => {
+          'id' => instance_of(Integer),
+          'email' => 'changed@sumofus.org',
+          'first_name' => 'Max',
+          'last_name' => 'Testy-Smith',
+          'country' => 'United Kingdom',
+          'postal' => 'EC2 1AB'
+        })
+      end
+    end
 
-            put '/api/member_services/members/', params: params, headers: valid_headers
-            expect(response.status).to eq 404
-            expect(json_hash.with_indifferent_access).to match(error_body)
-          end
-        end
+    context 'given a member with the passed email address does not exist' do
+      let(:params) do
+        {
+          email: 'new_member@sumofus.org',
+          member: {
+            first_name: 'Max'
+          }
+        }
+      end
 
-        context 'when the update fails' do
```

Comment:
Why did you remove error handling from when an update just fails?

Reply:
It's still there, but rewrote it to actually hit the service instead of mocking the response.